### PR TITLE
feat: add s3tables catalog

### LINKED ISSUE #754: Add s3tables catalog support
AWS announced a new product called S3 Tables, which allows using a bucket as an Iceberg catalog: https://aws.amazon.com/about-aws/whats-new/2024/12/amazon-s3-tables-apache-iceberg-tables-analytics-workloads/

The API can be found at: https://github.com/aws/aws-sdk-java-v2/blob/master/services/s3tables/src/main/resources/codegen-resources/service-2.json

It should be possible for iceberg-rust to support such a catalog.

### PR DESCRIPTION
**Pull request by flaneur2020**
fixes #754

this PR adds the implementation of s3tables catalog. i've tested the CRUD of namespaces/tables in my local laptop with a real s3tables bucket.

need to add a mocked test suites in ci in another pr.

### CODE REVIEW

**Comment by Xuanwo** [comment]
Hi @flaneur2020, I suggest splitting this PR into multiple ones to make it easier to review and accelerate the iteration speed.

**Comment by flaneur2020** [comment]
@Xuanwo i believe the rest part of this pr is adding tests, i've created a real s3tables bucket to test it and it looks work fine, can you give some suggestions about the test part?

i found the glue catalog used a mock service from `moto`, but the mock service for s3tables still not available in `moto` yet.

**Review by Xuanwo — COMMENTED** [review]
Thanks a lot for this PR, really great! Only some small suggestions.

File: crates/catalog/s3tables/src/catalog.rs [hunk 1]
```diff
@@ -0,0 +1,620 @@
...
+use iceberg::{
+    Catalog, Error, ErrorKind, Namespace, NamespaceIdent, Result, TableCommit, TableCreation,
+    TableIdent,
+};
+
+use crate::utils::{create_metadata_location, create_sdk_config};
+
+#[derive(Debug)]
+pub struct S3TablesCatalogConfig {
+    table_bucket_arn: String,
```
[the remark this answers is not in the document]
Comment (flaneur2020) [reply crates/catalog/s3tables/src/catalog.rs:39]: s3 tables bucket is very strange, no file is every stored in path under `s3://{s3table_bucket}/`, but every table has a special "bucket name" like `s3://{xxxxxxxx}_s3_table`. in the admin console of s3, you can not browse any of the files in this s3table bucket.

in my understanding, this ARN is the identifier of the the abstract s3tables, and it's used everywhere in the s3tables sdk, the plain bucket path is almost useless to us.

let me add it in the comments.

**Comment by flaneur2020** [comment]
@Xuanwo added comments on all the public apis, PTAL

**Review by Xuanwo — APPROVED** [review]
Really nice work, thank you!

**Pull request merged**